modules/articles/server/policies/articles.server.policy.js

```diff
@@ -20,6 +20,9 @@ exports.invokeRolesPolicies = function () {
     }, {
       resources: '/api/articles/:articleId',
       permissions: '*'
+    }, {
+      resources: '/api/parameterized-query/articles',
```

Comment:
I see what you mean, but I definitely prefer `/api/articles/:articleId`. It's more standard. The reason why I've used `parameterized-query/articles` is for two reasons. The first, you already pointed out, that there's a conflict with the `/:articleId` route, if I were to use `articles/query`. The second, is that any other collections that we may want to expose this feature to, could use the same pattern.

We may end up with something like:
`/api/parameterized-query/articles`
`/api/parameterized-query/users`
`/api/parameterized-query/some-other-data-collection`

I can see this type of API route configuration as a benefit, even if this *may* seem like a band-aid to a problem with how we're declaring our routes; given that we could have these conflicts. We can definitely look at our router configuration & solve the issue. But it's not something I wanted to do in this PR.